### PR TITLE
Uci 5

### PR DESCRIPTION
**Pull request by anmol-lf**
User Story: As a user, I want to prioritize units

I've faced some problems archiving the objects in arrays
So, I've used a work-around by archiving each unit object instead of the array and then de-archiving and appending each. This approach works but I don't think it is the proper way to do things and also increases unnecessary LOC. Please look through the code in DataHelper and PhysicalQuantities and provide any suggestions if you have.

### CODE REVIEW

File: unit-converter-ios-demo/unit-converter-ios-demo/controller/UnitTVC.swift [hunk 1]
```diff
@@ -20,6 +25,23 @@ class UnitTVC: UITableViewController {
...
     }
+    
+    override func tableView(tableView: UITableView, moveRowAtIndexPath fromIndexPath: NSIndexPath, toIndexPath: NSIndexPath) {
+        //var currentArray = physicalQuantities[selectedQuantityIndex].units
+        let itemToMove = physicalQuantities[selectedQuantityIndex].units[fromIndexPath.row]
+        physicalQuantities[selectedQuantityIndex].units.removeAtIndex(fromIndexPath.row)
+        physicalQuantities[selectedQuantityIndex].units.insert(itemToMove, atIndex: toIndexPath.row)
+        archive()
+        //print(currentArray == physicalQuantities[selectedQuantityIndex].units)
+    }
```
Comment (anmol-lf) [inline unit-converter-ios-demo/unit-converter-ios-demo/controller/UnitTVC.swift]: I thought that arrays were referenced, if so why does line 35 show false when line 30 is uncommented? Shouldnt they be the same?

File: unit-converter-ios-demo/unit-converter-ios-demo/model/PhysicalQuantity.swift [hunk 2]
```diff
@@ -1,21 +1,55 @@
...
+        self.unitReference = aDecoder.decodeObjectForKey("unitReference") as! Float
+    }
+    
+    func encodeWithCoder(aCoder: NSCoder) {
+        aCoder.encodeObject(unitName, forKey: "unitName")
+        aCoder.encodeObject(unitReference, forKey: "unitReference")
+    }
 }
 
 let length = PhysicalQuantity(name: "Length", unitDictionary: lengthUnits)
```
Comment (leapfrog-saugat) [inline unit-converter-ios-demo/unit-converter-ios-demo/model/PhysicalQuantity.swift]: Separate creation of Physical Quantities in new file.

File: unit-converter-ios-demo/unit-converter-ios-demo/controller/UnitTVC.swift [hunk 3]
```diff
@@ -5,6 +5,11 @@ class UnitTVC: UITableViewController {
         super.viewDidLoad()
     }
     
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        self.editing = true
```
Comment (leapfrog-saugat) [inline unit-converter-ios-demo/unit-converter-ios-demo/controller/UnitTVC.swift]: Avoid using self whenever possible.

File: unit-converter-ios-demo/unit-converter-ios-demo/helper/DataHelper.swift [hunk 4]
```diff
@@ -9,3 +12,20 @@ func retrieveData(withKey key: String) -> AnyObject? {
 func enterValue(object: AnyObject, forKey key: String){
     defaults.setObject(object, forKey: key)
 }
+
+func archive(){
+    let key = physicalQuantities[selectedQuantityIndex].quantityName
```
Comment (leapfrog-saugat) [inline unit-converter-ios-demo/unit-converter-ios-demo/helper/DataHelper.swift]: Change the name of the function to properly reflect its purpose. eg: archivePhysicalQuantities. Better yet create a generic function that takes both key and object as its parameter. eg: archive(object: NSObject, forKey key: String)

Comment (anmol-lf) [inline unit-converter-ios-demo/unit-converter-ios-demo/helper/DataHelper.swift]: :+1:

File: unit-converter-ios-demo/unit-converter-ios-demo/controller/UnitTVC.swift (hunk 3 above)
Comment (anmol-lf) [inline unit-converter-ios-demo/unit-converter-ios-demo/controller/UnitTVC.swift]: :+1:

File: unit-converter-ios-demo/unit-converter-ios-demo/model/PhysicalQuantity.swift (hunk 2 above)
Comment (anmol-lf) [inline unit-converter-ios-demo/unit-converter-ios-demo/model/PhysicalQuantity.swift]: :+1:

File: unit-converter-ios-demo/unit-converter-ios-demo/model/Unit.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import Foundation
+
+class Unit: NSObject, NSCoding {
+    var unitName: String
+    var unitReference: Float
+    
+    init(unitName: String, unitReference: Float){
+        self.unitName = unitName
+        self.unitReference = unitReference
```
Comment (leapfrog-saugat) [inline unit-converter-ios-demo/unit-converter-ios-demo/model/Unit.swift]: Remove self. Change local variable unitName to aUnitName.

File: unit-converter-ios-demo/unit-converter-ios-demo/model/Unit.swift [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    
+    required init(coder aDecoder: NSCoder) {
+        self.unitName = aDecoder.decodeObjectForKey("unitName") as! String
+        self.unitReference = aDecoder.decodeObjectForKey("unitReference") as! Float
+    }
+    
+    func encodeWithCoder(aCoder: NSCoder) {
+        aCoder.encodeObject(unitName, forKey: "unitName")
+        aCoder.encodeObject(unitReference, forKey: "unitReference")
+    }
```
Comment (leapfrog-saugat) [inline unit-converter-ios-demo/unit-converter-ios-demo/model/Unit.swift]: Create a separate file for string or other constants. Avoid the literal value "unitName" and use a variable instead.

File: unit-converter-ios-demo/unit-converter-ios-demo/helper/DataHelper.swift [hunk 7]
```diff
@@ -1,6 +1,9 @@
 import Foundation
 
 private let defaults = NSUserDefaults.standardUserDefaults()
+private let filemgr = NSFileManager.defaultManager()
+private let dirPaths = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)
+private let docsDir = dirPaths[0] as NSString
```
Comment (leapfrog-saugat) [inline unit-converter-ios-demo/unit-converter-ios-demo/helper/DataHelper.swift]: Avoid using filemgr or any such shortcuts. Try to be as verbose as possible while naming variables.

File: unit-converter-ios-demo/unit-converter-ios-demo/model/Unit.swift (hunk 5 above)
Comment (anmol-lf) [inline unit-converter-ios-demo/unit-converter-ios-demo/model/Unit.swift]: :+1:

File: unit-converter-ios-demo/unit-converter-ios-demo/model/Unit.swift (hunk 6 above)
Comment (anmol-lf) [inline unit-converter-ios-demo/unit-converter-ios-demo/model/Unit.swift]: :+1:

File: unit-converter-ios-demo/unit-converter-ios-demo/helper/DataHelper.swift (hunk 7 above)
Comment (anmol-lf) [inline unit-converter-ios-demo/unit-converter-ios-demo/helper/DataHelper.swift]: :+1:

**Pull request merged**